tools/RepositoryManagement/DevHome.RepositoryManagement/Services/EnhanceRepositoryService.cs

```diff
@@ -0,0 +1,170 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using DevHome.Common.Services;
+using DevHome.Common.TelemetryEvents.RepositoryManagement;
+using DevHome.Common.Windows.FileDialog;
+using DevHome.Customization.ViewModels;
+using DevHome.Telemetry;
+using FileExplorerSourceControlIntegration;
+using Microsoft.UI.Xaml;
+using Microsoft.Windows.DevHome.SDK;
+using Serilog;
+using Windows.Foundation.Collections;
+using Windows.Storage;
+
+namespace DevHome.RepositoryManagement.Services;
+
+public class EnhanceRepositoryService
+{
+    private const string ErrorEventName = "DevHome_EnhanceRepositoryError_Event";
+
+    private readonly ILogger _log = Log.ForContext("SourceContext", nameof(EnhanceRepositoryService));
+
+    private readonly FileExplorerViewModel _sourceControlRegistrar;
+
+    private readonly Window _window;
+
+    private readonly ITelemetry _telemetry;
+
+    private readonly List<IExtensionWrapper> _repositorySourceControlProviders;
+
+    public EnhanceRepositoryService(
+        FileExplorerViewModel sourceControlRegistrar,
+        Window window,
+        IExtensionService extensionService)
+    {
+        _sourceControlRegistrar = sourceControlRegistrar;
+        _telemetry = TelemetryFactory.Get<ITelemetry>();
+        _window = window;
+        _repositorySourceControlProviders = extensionService.GetInstalledExtensionsAsync(ProviderType.LocalRepository).Result.ToList();
+    }
+
+    /// <summary>
+    /// Gets the location of an existing repository and hook up the path to both
+    /// the source control provider and file explorer.
+    /// </summary>
+    /// <returns>Path to the new repository.  String.Empty for any errors</returns>
+    public async Task<(string RepositoryLocation, Guid sourceControlClassId)> SelectRepositoryAndMakeItEnhanced()
+    {
+        var maybeRepositoryPath = await GetRepositoryLocationFromUser();
```

Comment:
>GetRepositoryLocationFromUser [](http://example.com/codeflow?start=40&length=29)

is it weird that this class pops a UI dialog when it doesn't have anything else to do with UI?